sounddevice.py

```
@@ -2265,10 +2265,66 @@ def reset(self):
+
+    Parameters
+    ----------
+    msg : optional
+        User error message.
+    errcode : int or None, optional
+        PaErrorCode value or None if there is no PaErrorCode associated
+        with the error.
+
+    Notes
+    -----
+    - Some PortAudioError's do not have an associated PaErrorCode. In
+      such a case `paerrcode` will be None.
+
+    - The `hostapi`, `hosterrcode`, and `hosterrmsg` attributes will be
+      None except for the case when `paerrcode` ==
+      `paUnanticipatedHostError`.
+    
+    Attributes
+    ----------
+    paerrcode : int or None
+        The PaErrorCode integer value or None if there is no associated
+        error code.
+    hostapi : str or None
+        The name of the host API which returned the error code.
+    hosterrcode : int or None
+        The host error code returned.
+    hosterrmsg : str or None
+        A textual description of the host error if available, otherwise
+        a zero-length string.
+    """
+    def __init__(self, msg=None, errcode=None):
+        self.paerrcode = errcode
+        self.hosterrcode = self.hostapi = self.hosterrmsg = None
+
+        if errcode:
+            errmsg = "[PaErrorCode {0:d}] ".format(self.paerrcode)
+            if errcode == _lib.paUnanticipatedHostError:
+                info = _lib.Pa_GetLastHostErrorInfo()
+                hostapi = _lib.Pa_HostApiTypeIdToHostApiIndex(info.hostApiType)
+                apiinfo = _lib.Pa_GetHostApiInfo(hostapi)
+                self.hosterrcode = info.errorCode
+                self.hostapi = _ffi.string(apiinfo.name).decode() if apiinfo else 'unknown'
+                self.hosterrmsg = _ffi.string(info.errorText).decode()
+                errmsg += "Unanticipated {0!r} host error".format(self.hostapi)
+            else:
+                errmsg += _ffi.string(_lib.Pa_GetErrorText(errcode)).decode()
+            if msg is not None:
+                errmsg += ": {0!s}".format(msg)
+            elif self.hosterrmsg:
```

Why are you ignoring the error message if a `msg` is given?